iothub_client/samples/pnp/pnp_temperature_controller/pnp_temperature_controller.c

```diff
@@ -276,184 +327,184 @@ static void PnP_TempControlComponent_DeviceTwinCallback(DEVICE_TWIN_UPDATE_STATE
+    unsigned char* serializedProperties = NULL;
+    size_t serializedPropertiesLength;
 
-    if ((jsonToSend = PnP_CreateReportedProperty(NULL, g_serialNumberPropertyName, g_serialNumberPropertyValue)) == NULL)
+    // The first step of reporting properties is to serialize it into an IoT Hub friendly format.  You can do this by either
+    // implementing the PnP convention for building up the correct JSON or more simply to use IoTHubClient_Serialize_ReportedProperties.
+    if ((clientResult = IoTHubClient_Serialize_ReportedProperties(&property, 1, NULL, &serializedProperties, &serializedPropertiesLength)) != IOTHUB_CLIENT_OK)
     {
-        LogError("Unable to build serial number property");
+        LogError("Unable to serialize reported state, error=%d", clientResult);
     }
-    else
+    // The output of IoTHubClient_Serialize_ReportedProperties is sent to IoTHubDeviceClient_LL_SendPropertiesAsync to perform network I/O.
+    else if ((clientResult = IoTHubDeviceClient_LL_SendPropertiesAsync(deviceClient, serializedProperties, serializedPropertiesLength, NULL, NULL)) != IOTHUB_CLIENT_OK)
     {
-        const char* jsonToSendStr = STRING_c_str(jsonToSend);
-        size_t jsonToSendStrLen = strlen(jsonToSendStr);
-
-        if ((iothubClientResult = IoTHubDeviceClient_LL_SendReportedState(deviceClient, (const unsigned char*)jsonToSendStr, jsonToSendStrLen, NULL, NULL)) != IOTHUB_CLIENT_OK)
-        {
-            LogError("Unable to send reported state, error=%d", iothubClientResult);
-        }
-        else
-        {
-            LogInfo("Sending serialNumber property to IoTHub");
-        }
+        LogError("Unable to send reported state, error=%d", clientResult);
     }
 
-    STRING_delete(jsonToSend);
+    IoTHubClient_Serialize_Properties_Destroy(serializedProperties);
 }
 
 //
-// GetConnectionStringFromEnvironment retrieves the connection string based on environment variable
+// CreateDeviceClientLLHandle does the creates the IOTHUB_DEVICE_CLIENT_LL_HANDLE based on environment configuration.
+// If PNP_CONNECTION_SECURITY_TYPE_DPS is used, the call will block until DPS provisions the device.
 //
-static bool GetConnectionStringFromEnvironment()
+static IOTHUB_DEVICE_CLIENT_LL_HANDLE CreateDeviceClientLLHandle(void)
 {
-    bool result;
+    IOTHUB_DEVICE_CLIENT_LL_HANDLE deviceClient = NULL;
 
-    if ((g_pnpDeviceConfiguration.u.connectionString = getenv(g_connectionStringEnvironmentVariable)) == NULL)
+    if (g_pnpDeviceConfiguration.securityType == PNP_CONNECTION_SECURITY_TYPE_CONNECTION_STRING)
     {
-        LogError("Cannot read environment variable=%s", g_connectionStringEnvironmentVariable);
-        result = false;
+        if ((deviceClient = IoTHubDeviceClient_LL_CreateFromConnectionString(g_pnpDeviceConfiguration.u.connectionString, MQTT_Protocol)) == NULL)
+        {
+            LogError("Failure creating IotHub client.  Hint: Check your connection string");
```

Comment:
```suggestion
            LogError("Failure creating Iot Hub client.  Hint: Check your connection string");
```

Reply:
Fixing here and the other places in logs/comments that this should've been separate.